src/main/java/com/powsybl/openloadflow/network/util/GenerationActivePowerDistributionStep.java

```diff
@@ -134,8 +137,11 @@ private double getParticipationFactor(LfGenerator generator) {
     }
 
     private boolean isParticipating(LfGenerator generator) {
-        // check first if generator is set to be participating
-        if (!generator.isParticipating()) {
+        // check first if generator:
+        // - is set to be participating
+        // - and if option is set to distribute only on generators on voltage control
+        if (!generator.isParticipating() ||
```

Comment:
At each distribution outer loop, the controlled buses are changing (PV buses) so are the associated controller buses too. If you want a dynamic behavior, you must prefer using the method `isGeneratorVoltageControlEnabled()` from the controller bus (even if in case of remote voltage control? It is not the PV bus...). If you don't want a dynamic behavior, I think you can use the type (but without a dedicated method called `hasVoltageControl' because we have too much and we have tried to make the difference between voltage control, controllers and controlled buses).

Reply:
I don't want dynamic behavior. Also I fear that doing this would result in "distributed slack" and "reactive limits" fighting each other.

But checking again my requirements, the `hasVoltageControl` as currently implemented is not what I need functionally speaking, i need usage of the "raw" voltageRegulatorOn generator status.

Reply:
So, maybe you have to add a new type because "OFF" can mean off in input data and off because checks are not passing.

Reply:
yes exactly, this is why I wrote https://github.com/powsybl/powsybl-open-loadflow/pull/1063#discussion_r1673546364 ...

but I am wondering if this is a feature we want in OLF or if we want to leave this feature outside ... it is quite easy to disable slack distribution on a generator via ActivePowerControl extension ...

Reply:
Sorry I miss that, yes if you only need IIDM "voltageRegulatorOn" status, better try first to use active power control extension.